### PR TITLE
Firsts tests about primitive add

### PR DESCRIPTION
**Pull request by BastienDegardins**
Add tests for add with overflow and without overflow

### CODE REVIEW

**Comment by guillep** [comment]
Tests are green for both 32 and 64 bits! good!

image
image

File: smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st [hunk 1]
```diff
@@ -66,6 +66,69 @@ VMPrimitiveTest >> tearDown [
 	super tearDown
 ]
 
+{ #category : #'tests - primitiveAtPut' }
+VMPrimitiveTest >> testPrimitiveAddWithNoOverflow [
+
+	interpreter push: (memory integerObjectOf: 42). "Marker"
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st:72]: The marker is not needed in this test, right?

File: smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st [hunk 2]
```diff
@@ -66,6 +66,69 @@ VMPrimitiveTest >> tearDown [
...
+{ #category : #'tests - primitiveAtPut' }
+VMPrimitiveTest >> testPrimitiveAddWithNoOverflowPopsOperands [
+
+	interpreter push: (memory integerObjectOf: 42). "Marker"
+	interpreter push: (memory integerObjectOf: 1).
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveAdd. 
+	
+	interpreter popStack.
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st:94]: Maybe we should add a comment to explain that we are popping the result of 1+2, so the top of the stack should now contain the marker object.

File: smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st [hunk 3]
```diff
@@ -66,6 +66,69 @@ VMPrimitiveTest >> tearDown [
...
+	
+	
+]
+
+{ #category : #'tests - primitiveAtPut' }
+VMPrimitiveTest >> testPrimitiveAddWithOverflowPreservesStack [
+	
+	| maxSmallInt |
+	
+	maxSmallInt:= (memory integerObjectOf: (memory maxSmallInteger)).
```
Comment (guillep) [inline smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st:119]: The parenthesis around "memory maxSmallInteger" are not necessary ;)

**Comment by guillep** [comment]
@BastienDegardins I did some comments, I'm gonna merge the PR, we can fix the comments in the next PR ;)
Good job! :)
